interface.go

```diff
@@ -85,7 +85,7 @@ type GlobalLookupFactory interface {
 	// Help text for the CLI
 	Help() string
 	// Return a single scanner which will scan a single host
-	MakeRoutineFactory() (RoutineLookupFactory, error)
+	MakeRoutineFactory(int) (RoutineLookupFactory, error)
```

Comment:
Please run go lint on your code and fix some of the style errors, for example write comments following convention.